### PR TITLE
feat(wallet): Max Slippage Modal

### PR DESCRIPTION
**Pull request by Douglashdaniel**
## Description 

- Introduces a new `Max Slippage` modal to allow user to set the `Slippage Tolerance` for a `Swap` or `Bridge` transaction.
- Re-sorts the `Quote Info` to give `Slippage` higher priority in the display.

Resolves <https://github.com/brave/brave-browser/issues/39514>

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
1. Open the `Wallet` and navigate to the `Swap` or `Bridge` screen.
2. Select some tokens and get a quote.
3. Once the `Quote Info` appears click on the `Max slippage` button
4. Test changing between `Suggested` and `Custom` slippage.
5. After you update the `slippage`, the quote should automatically be refreshed and reflect the `Slippage` amount you entered.

https://github.com/brave/brave-core/assets/40611140/072bc847-32a8-4bf9-875f-ccb0cf99b5f1

### CODE REVIEW

File: components/brave_wallet_ui/page/screens/swap/components/swap/max_slippage/max_slippage.style.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+// Shared Styles
+import {
+  WalletButton //
+} from '../../../../../../components/shared/style'
+
+export const OptionButton = styled(WalletButton)<{
+  isSelected: boolean
+}>`
+  --radio-icon-color: ${(p) =>
```
Comment (josheleonard) [inline components/brave_wallet_ui/page/screens/swap/components/swap/max_slippage/max_slippage.style.ts:18]: can we move this to the RadioIcon for clarity?

File: components/brave_wallet_ui/page/screens/swap/components/swap/max_slippage/max_slippage.tsx [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+  LeoSquaredButton,
+  Row,
+  Text,
+  VerticalSpace
+} from '../../../../../../components/shared/style'
+
+const suggestedSlippage = '0.5'
+const mainstreamAssetPairsSlippage = '0.5'
+const stablecoinPairsSlippage = '0.5'
+const othersSlippage = '1'
```
Comment (josheleonard) [inline components/brave_wallet_ui/page/screens/swap/components/swap/max_slippage/max_slippage.tsx:30]: nit: can we use UPPER_SNAKE_CASE to help convey that these are constants?

File: components/brave_wallet_ui/page/screens/swap/components/swap/max_slippage/max_slippage.tsx [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+        >
+          <Column alignItems='flex-start'>
+            <Text
+              isBold={true}
+              textSize='12px'
+              textColor='primary'
+            >
+              {getLocale('braveWalletSuggested')}
+            </Text>
+            <VerticalSpace space='8px' />
```
Comment (josheleonard) [inline components/brave_wallet_ui/page/screens/swap/components/swap/max_slippage/max_slippage.tsx:147]: can we set the column gap instead?

File: components/brave_wallet_ui/page/screens/swap/components/swap/max_slippage/max_slippage.tsx [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+        >
+          <Column alignItems='flex-start'>
+            <Text
+              isBold={true}
+              textSize='12px'
+              textColor='primary'
+            >
+              {getLocale('braveWalletCustom')}
+            </Text>
+            <VerticalSpace space='8px' />
```
Comment (josheleonard) [inline components/brave_wallet_ui/page/screens/swap/components/swap/max_slippage/max_slippage.tsx:176]: Can we use the column gap instead?

File: components/brave_wallet_ui/page/screens/swap/components/swap/quote-info/quote-info.tsx [hunk 5]
```diff
@@ -344,174 +367,192 @@ export const QuoteInfo = (props: Props) => {
...
+                    padding='8px 0px 0px 0px'
+                  >
+                    {selectedQuoteOption.sources.map((source, idx) => (
+                      <Row
+                        key={source.name}
+                        width='unset'
+                      >
+                        {getLPIcon(source) !== '' ? (
+                          <LPIcon icon={getLPIcon(source)} />
+                        ) : null}
```
Comment (josheleonard) [inline components/brave_wallet_ui/page/screens/swap/components/swap/quote-info/quote-info.tsx:498]: perf: let's compute `getLPIcon(source)` once per rendered item

File: components/brave_wallet_ui/page/screens/swap/components/swap/quote-info/quote-info.tsx [hunk 6]
```diff
@@ -565,6 +606,77 @@ export const QuoteInfo = (props: Props) => {
...
+                    <>
+                      <BraveFeeDiscounted
+                        textSize='12px'
+                        isBold={true}
+                        textColor='primary'
+                      >
+                        {braveFee.feePct}%
+                      </BraveFeeDiscounted>
+
+                      {effectiveFeeAmount && effectiveFeeAmount.gt(0) && (
```
Comment (josheleonard) [inline components/brave_wallet_ui/page/screens/swap/components/swap/quote-info/quote-info.tsx:646]:
```suggestion
                      {effectiveFeeAmount?.gt(0) && (
```

File: components/brave_wallet_ui/page/screens/swap/hooks/useSwap.ts [hunk 7]
```diff
@@ -434,7 +426,11 @@ export const useSwap = () => {
         provider:
           overrides.provider === undefined
             ? selectedProvider
-            : overrides.provider
+            : overrides.provider,
+        slippage:
+          overrides.slippage === undefined
+            ? slippageTolerance
+            : overrides.slippage
```
Comment (josheleonard) [inline components/brave_wallet_ui/page/screens/swap/hooks/useSwap.ts:433]: Will this work?

```suggestion
        slippage: overrides.slippage ?? slippageTolerance
```

File: components/brave_wallet_ui/page/screens/swap/components/swap/max_slippage/max_slippage.style.ts (hunk 1 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/page/screens/swap/components/swap/max_slippage/max_slippage.style.ts]: Moved!

File: components/brave_wallet_ui/page/screens/swap/components/swap/max_slippage/max_slippage.tsx (hunk 2 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/page/screens/swap/components/swap/max_slippage/max_slippage.tsx]: Changed to `UPPER_SNAKE_CASE`!

File: components/brave_wallet_ui/page/screens/swap/components/swap/max_slippage/max_slippage.tsx (hunk 4 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/page/screens/swap/components/swap/max_slippage/max_slippage.tsx]: Done!

File: components/brave_wallet_ui/page/screens/swap/components/swap/max_slippage/max_slippage.tsx (hunk 3 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/page/screens/swap/components/swap/max_slippage/max_slippage.tsx]: Done!

File: components/brave_wallet_ui/page/screens/swap/components/swap/quote-info/quote-info.tsx (hunk 5 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/page/screens/swap/components/swap/quote-info/quote-info.tsx]: Done!

File: components/brave_wallet_ui/page/screens/swap/components/swap/quote-info/quote-info.tsx (hunk 6 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/page/screens/swap/components/swap/quote-info/quote-info.tsx]: Updated!

File: components/brave_wallet_ui/page/screens/swap/hooks/useSwap.ts (hunk 7 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/page/screens/swap/hooks/useSwap.ts]: Yep works, changed!

**Review by josheleonard — APPROVED** [review]

**Review by yrliou — APPROVED** [review]
core++

**Review by mkarolin — APPROVED** [review]
`strings`++

**Pull request merged**